bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/CMakeBuildTab2.java

```diff
@@ -340,4 +352,25 @@ public String getName()
 		return "CMake"; //$NON-NLS-1$
 	}
 
+	private String getCmakeArgumentsWithAbsProjectPath(IProject project, String cmakeArgumets)
+	{
+		String[] cmakeArgsArr = cmakeArgsText.getText().trim().replaceAll(" +", " ").split(" "); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+		String buildFolder = StringUtil.EMPTY;
+		for (int i = 0; i < cmakeArgsArr.length; i++)
+		{
+			if (cmakeArgsArr[i].equals("-B") && i + 1 < cmakeArgsArr.length) //$NON-NLS-1$
+			{
+				buildFolder = cmakeArgsArr[i + 1];
+				break;
+			}
+		}
```

Comment:
Can we do this?

```
String buildFolder = StringUtil.EMPTY;
String[] cmakeArgsArr = cmakeArgsText.getText().trim().split("\\s+");  // Split on any whitespace

for (int i = 0; i < cmakeArgsArr.length - 1; i++) {
  if (cmakeArgsArr[i].equals("-B")) {
    buildFolder = cmakeArgsArr[i + 1];
    break;
  }
}
```

Reply:
Hi @kolipakakondal, thanks for the feedback. Fixed in the last commit